torch/csrc/autograd/variable.cpp

```diff
@@ -181,19 +181,29 @@ void rebase_history(const Variable& self, Edge gradient_edge) {
   }
 
   set_gradient_edge(self, std::move(gradient_edge));
+  if (self.retains_grad()) {
+    auto old_fn = self.grad_fn()->next_edges()[0].function;
```

Comment:
Why not do that before the grad_fn() is updated? That way you don't have to traverse the edge.

Reply:
Good catch! This would probably simplify the in-place over view case as well